src/Microsoft.Identity.Client/ITokenCache.cs

```diff
@@ -47,23 +48,64 @@ namespace Microsoft.Identity.Client
     /// </summary>
     public interface ITokenCache
     {
-#if !ANDROID_BUILDTIME && !iOS_BUILDTIME 
+#if !ANDROID_BUILDTIME && !iOS_BUILDTIME
+
+        /// <summary>
+        /// Sets a function to be notified before any library method accesses the cache. This gives an option to the
+        /// delegate to deserialize a cache entry for the application and accounts specified in the <see cref="TokenCacheNotificationArgs"/>.
+        /// See https://aka.ms/msal-net-token-cache-serialization
+        /// </summary>
+        /// <param name="beforeAccess">Function set in order to handle the cache deserialiation</param>
+        /// <remarks>In the case where the function is used to deserialize the cache, it might
+        /// want to call <see cref="Deserialize(byte[])"/></remarks>
+        void SetAsyncBeforeAccess(Func<TokenCacheNotificationArgs, Task> beforeAccess);
```

Comment:
do we want to add in the comments:
See also <see cref="M:SetAsyncBeforeAccess(TokenCacheNotificationArgs"/> which lets you add a callback called synchronously?

We also want to modify the XML comments of the non-async methods to point to the async?

Reply:
In term of naming. what do you think of SetBeforeAccessAsyncCallback?